### ISSUE TITLE
TapeCart isn't working

### ISSUE THREAD

**Issue by markusC64** [GideonZ/1541ultimate#275]
TapeCart isn't working

Well, this is not a recent regression. It's not working for a longer time.

This is just an issue to track it.

**Comment by markusC64** [comment]
TapeCart SD does not work on U64, too.

**Comment by GideonZ** [comment]
It's a bit of a choice. It's either:
a) pass the 6510 PIO compliance test of Vice test suite, or
b) let peripherals like TapeCart work.

Maybe it should be configurable.

**Comment by markusC64** [comment]
> It's a bit of a choice. It's either: a) pass the 6510 PIO compliance test of Vice test suite, or b) let peripherals like TapeCart work. Maybe it should be configurable.

Good idea. At least until there's a better solution.

**Comment by GideonZ** [comment]
Unfortunately, there is not really a better solution. You see, the PIO test
in the test suite expects some capacitance on the I/O pin. So when the pin
is set to input, the value should linger there for 'x' time. However, it is
*unknown* whether an external device drives the pin in the mean time. So in
order to really fix it, one would have to add a small capacitor and
resistor to the I/O pin on the FPGA and then pass the external signal
through. Right now, the capacitance emulator is attached to the input, not
the external signal.

**Comment by mrdudz** [comment]
> It's a bit of a choice. It's either:
> a) pass the 6510 PIO compliance test of Vice test suite, or
> b) let peripherals like TapeCart work.

I don't agree. Tapecart works fine on real C64s (that pass those tests) afterall :)

**Comment by GideonZ** [comment]
This is not something to agree / disagree on. It is a consequence of how the hardware works, not something that can be fixed in the firmware. In this case, the preference was to have the vice test pass, but that is obviously the wrong choice.

**Comment by mrdudz** [comment]
Well, when the VICE test does not pass - you will see random (mostly older) software not working, because it relies on those things.

**Comment by GideonZ** [comment]
Then this should be a switchable option in the config.

**Comment by markusC64** [comment]
Yes, switchable config is good compared to the alternatives.

**Comment by TomCat666** [comment]
Any news on this issue? Would love to have the setting to enable the tapecart support on the tape port, or even better support .tcrt files with U64. Lots of tapecart releases coming out lately.
